dwave/optimization/generators.py

```diff
@@ -605,23 +605,19 @@ def job_shop_scheduling(times: numpy.typing.ArrayLike, machines: numpy.typing.Ar
 
     # Ensure that for each job, its tasks do not overlap
     for j in range(num_jobs):
-        ends = end_times[j, :][machines[j, :-1]]
-        starts = start_times[j, :][machines[j, :]][1:]
+        ends = end_times[j, machines[j, :-1]]
+        starts = start_times[j, machines[j, 1:]]
+
         model.add_constraint((ends <= starts).all())
 
     # Ensure for each machine, its tasks do not overlap
     for m in range(num_machines):
         order = orders[m]
 
-        ends = end_times[:, m]
```

Comment:
now the question is: here we were using one single basic indexing and reused that all over. Unless I'm missing something, we are increasing the number of nodes, right?

Reply:
I guess we could save `order[:-1]` and `order[1:]` if we wanted to... we really just need to add node deduplication as a presolve step. Maybe I'll look at that.